riptide_imu:
  ros__parameters:
    port: "/dev/ttyUSB0"
    baud: 115200
    reconnect_ms: 500
    # Async Output Type
    # Set to VNOFF because output is handled by BOR
    AsyncDataOutputType: 0
    # In Hz
    AsyncDataOutputFrequency: 20
    # Sync Control
    syncInMode: 3                    # SYNCINMODE_COUNT
    syncInedge: 0                    # SYNCINEDGE_RISING
    syncInSkipFactor: 0
    syncOutMode: 0                   # SYNCOUTMODE_NONE
    syncOutPolarity: 0               # SYNCOUTPOLARITY_NEGATIVE
    syncOutSkipFactor: 0
    syncOutPulseWidth_ns: 100000000
    # Communication protocol control
    serialCount: 0                   # COUNTMODE_NONE
    serialStatus: 0                  # STATUSMODE_OFF
    spiCount: 0                      # COUNTMODE_NONE
    spiStatus: 0                     # STATUSMODE_OFF
    serialChecksum: 1                # CHECKSUMMODE_CHECKSUM
    spiChecksum: 0                   # CHECKSUMMODE_OFF
    errorMode: 1                     # ERRORMODE_SEND
    # Binary output register 1
    BO1:
      asyncMode: 3                  # ASYNCMODE_BOTH
      rateDivisor: 40               # 20 Hz
      commonField: 0x7FFF           # THIS MAKES IT WORK FOR SOME REASON
      timeField: 0x000              # TIMEGROUP_NONE
      imuField: 0x000               # IMUGROUP_NONE
      # set gpsField directly in source to enforce or condition
      attitudeField: 0x000          # ATTITUDEGROUP_NONE
      # set insField directly in source to enforce or condition
      gps2Field: 0x00               # GPSGROUP_NONE
    # Covariance data
    orientation_covariance: [0.01,  0.0,   0.0,
                            0.0,   0.01,  0.0,
                            0.0,   0.0,   0.01]
    angular_velocity_covariance: [0.01,  0.0,   0.0,
                         0.0,   0.01,  0.0,
                         0.0,   0.0,   0.01]
    linear_acceleration_covariance: [0.01,  0.0,   0.0,
                         0.0,   0.01,  0.0,
                         0.0,   0.0,   0.01]

    frame_id: "vectornav"
